tests/test_transform.py

```diff
@@ -0,0 +1,157 @@
+    trans = _SearchSpaceTransform({"x0": distribution})
+    trans_params = trans.transform({"x0": param})
+
+    if isinstance(distribution, CategoricalDistribution):
+        expected_bounds_shape = (len(distribution.choices), 2)
+        expected_params_shape = (len(distribution.choices),)
+    else:
+        expected_bounds_shape = (1, 2)
+        expected_params_shape = (1,)
+    assert trans.bounds.shape == expected_bounds_shape
+    assert trans.bounds.dtype == numpy.float64
+    assert trans_params.shape == expected_params_shape
+    assert trans_params.dtype == numpy.float64
+
+
+@pytest.mark.parametrize("transform_log", [True, False])
+@pytest.mark.parametrize("transform_step", [True, False])
+@pytest.mark.parametrize(
+    "param,distribution",
+    [
+        (0, IntUniformDistribution(0, 3)),
+        (1, IntLogUniformDistribution(1, 10)),
+        (2, IntUniformDistribution(0, 10, step=2)),
+        (0.0, UniformDistribution(0, 3)),
+        (1.0, LogUniformDistribution(1, 10)),
+        (0.2, DiscreteUniformDistribution(0, 1, q=0.2)),
+    ],
+)
+def test_search_space_transform_numerical(
+    transform_log: bool,
+    transform_step: bool,
+    param: Any,
+    distribution: BaseDistribution,
+) -> None:
+    trans = _SearchSpaceTransform({"x0": distribution}, transform_log, transform_step)
+
+    expected_low = distribution.low  # type: ignore
+    expected_high = distribution.high  # type: ignore
+
+    if isinstance(distribution, LogUniformDistribution):
+        if transform_log:
+            expected_low = math.log(expected_low)
+            expected_high = math.log(expected_high)
+    elif isinstance(distribution, DiscreteUniformDistribution):
+        if transform_step:
+            half_step = 0.5 * distribution.q
+            expected_low -= half_step
+            expected_high += half_step
+    elif isinstance(distribution, IntUniformDistribution):
+        if transform_step:
+            half_step = 0.5 * distribution.step
+            expected_low -= half_step
+            expected_high += half_step
+    elif isinstance(distribution, IntLogUniformDistribution):
+        if transform_step:
+            half_step = 0.5
+            expected_low -= half_step
+            expected_high += half_step
+        if transform_log:
+            expected_low = math.log(expected_low)
+            expected_high = math.log(expected_high)
+
+    for bound in trans.bounds:
+        assert bound[0] == expected_low
+        assert bound[1] == expected_high
+
+    trans_params = trans.transform({"x0": param})
+    assert trans_params.size == 1
+
+    if isinstance(distribution, (IntUniformDistribution, IntLogUniformDistribution)):
+        assert expected_low <= trans_params <= expected_high
+    else:
+        # TODO(hvy): Change second `<=` to `<` when `suggest_float` is fixed.
+        assert expected_low <= trans_params <= expected_high
```

Comment:
[just a question] What should we do to address this TODO comment?

Reply:
Noticed it's already addressed in the sense that `trans_params` is hard-coded instead of being suggested by a study. Fixed the code and removed the comment. Thanks!